core/arch/arm/plat-rockchip/main.c

```diff
@@ -14,14 +14,15 @@
 #include <platform_config.h>
 #include <stdint.h>
 
-static struct gic_data gic_data;
+static struct gic_data gic_data __nex_bss;
 
 #if defined(CFG_EARLY_CONSOLE)
-static struct serial8250_uart_data early_console_data;
+static struct serial8250_uart_data early_console_data __nex_bss;
 register_phys_mem_pgdir(MEM_AREA_IO_NSEC,
 			CFG_EARLY_CONSOLE_BASE, CFG_EARLY_CONSOLE_SIZE);
 #endif
 
+register_dynamic_shm(CFG_NSEC_DDR_0_BASE, CFG_NSEC_DDR_0_SIZE);
```

Comment:
Please use `register_ddr()` instead.

Reply:
Yes, the change of  console baudrate is not necessary , it's just convenient for me to see the log.

The change of default TZDRAM and SHMEM addresses is the prerequisite for booting optee os.

As for register DDR,I make some changes that  using register_ddr() instead of register_dynamic_shm() as you suggested.

The __nex_bss part is exactly for enable virtualization